### PR TITLE
Issue 4038: Use cxf-rt-transports-http-hc module for client transport

### PR DESCRIPTION
**Pull request by andymc12**
This pulls in the cxf-rt-transports-http-hc module and recent versions
of the Apache HTTP Client.  These modules should enable faster and more
dynamic client requests from JAX-RS (and potentially MP Rest Client)
applications.

This adds new functionality, and should not be considered as a release bug.

This resolves issue #4038.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Fixed auto-feature name conflict and merge conflict with httpcommons-1.0 feature.  New #build

**Comment by andymc12** [comment]
updated to CXF 3.2.6.  new #build

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/component/AsyncClientRunnableWrapperManager.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.cxf.message.Message;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+import com.ibm.ws.jaxrs20.client.AsyncClientRunnableWrapper;
+
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/component/AsyncClientRunnableWrapperManager.java]: can remove this empty comment block

File: dev/com.ibm.ws.jaxrs.2.x.concurrent/findbugs.exclude.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!--
+    Copyright (c) 2017 IBM Corporation and others.
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.x.concurrent/findbugs.exclude.xml]: 2018

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.hc.3.2/src/org/apache/cxf/transport/http/asyncclient/AsyncHTTPConduitFactory.java [hunk 3]
```diff
@@ -0,0 +1,443 @@
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.hc.3.2/src/org/apache/cxf/transport/http/asyncclient/AsyncHTTPConduitFactory.java]: // Liberty Change?

**Comment by andymc12** [comment]
Resolved code review comments - new #build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
2 Failures - both unrelated to these changes.

**Pull request merged**